modules/oe_content_persistent/tests/Functional/PersistentUrlControllerTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_content_persistent\Functional;
+
+use Drupal\language\Entity\ConfigurableLanguage;
+use Drupal\node\Entity\Node;
+use Drupal\node\Entity\NodeType;
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * Test PersistentUrlController response with caching mechanism.
+ *
+ * @group oe_content
+ */
+class PersistentUrlControllerTest extends BrowserTestBase {
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = [
+    'path',
+    'node',
+    'user',
+    'system',
+    'language',
+    'content_translation',
+    'oe_content_persistent',
+    'page_cache',
+    'dynamic_page_cache',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $node_type = NodeType::create(['type' => 'page']);
+    $node_type->save();
+
+    ConfigurableLanguage::create(['id' => 'fr'])->save();
+
+    $config = $this->config('language.negotiation');
+    $config->set('url.prefixes', ['en' => '', 'fr' => 'fr'])
+      ->save();
+  }
+
+  /**
+   * Tests the path cache with Persistent Controller.
+   */
+  public function testPersistentUrlController(): void {
+
+    $node_storage = \Drupal::entityTypeManager()->getStorage('node');
+
+    $node = Node::create([
+      'title' => 'Testing create()',
+      'type' => 'page',
+      'path' => ['alias' => '/foo'],
+      'status' => TRUE,
+      'uid' => 0,
+    ]);
+    $node->save();
+
+    $this->drupalGet('/content/' . $node->uuid());
+    $this->assertResponse(200);
+    $this->assertUrl('/foo');
+    $this->assertText('Testing create()');
+
+    $node->path->alias = '/foo2';
+    $node->save();
+
+    $this->drupalGet('/content/' . $node->uuid());
+    $this->assertResponse(200);
+    $this->assertUrl('/foo2');
+    $this->assertText('Testing create()');
+
+    $node->path->alias = '';
+    $node->save();
+
+    $this->drupalGet('/content/' . $node->uuid());
+    $this->assertResponse(200);
+    $this->assertUrl('/node/' . $node->id());
+    $this->assertText('Testing create()');
+
+    // Add a translation, verify it is being saved as expected.
+    $translation = $node->addTranslation('fr', $node->toArray());
+    $translation->get('path')->alias = '/petitbar';
+    $translation->save();
+
+    $this->drupalGet('/fr/content/' . $node->uuid());
```

Comment:
Putting the langcode in here might not be too good of an idea of we want to use the URI as the id of the content.

Reply:
I don't know yet how this is possible to avoid lang prefixes. Because in fact for instance node entity type have the same uuid for original and translated entity.

Reply:
To be able to explain further my argument I would like to point here https://www.w3.org/2001/tag/doc/identify
While reading this keep, keep the following assumptions I make in mind:
- A URI identities a resource
- A resource can be represented by a resource representation
- A translation of a node is in fact a different representation of the same resource
- We want a unique way of identifying resources (not resource representations)

I hope this somehow makes sense :)

Reply:
What we meant is that this module shouldn't care of redirecting to the proper language, as it's not the scope of the system. We will keep the linked language only in the ticket about inline content linking with something like
`<a href="/content/my-uuid" lang="fr">My french link</a>`. This will be resolved by the filter back to `/fr/my-french-link`.
If the body is consumed by another system that doesn't cannot resolve the purl to the alias, it will loose the language.